datalad_next/patches/fix_ria_ora_tests.py

```diff
@@ -0,0 +1,1010 @@
+                                                    path=store)
+        arch_url = "ria+ssh://{host}{path}".format(host=host,
+                                                   path=archiv_store)
+    else:
+        store_url = "ria+{}".format(store.as_uri())
+        arch_url = "ria+{}".format(archiv_store.as_uri())
+
+    create_store(io, ppp_store, '1')
+
+    # TODO: Re-establish test for version 1
+    # version 2: dirhash
+    create_ds_in_store(io, ppp_store, ds.id, '2', '1')
+
+    # add special remote
+    init_opts = common_init_opts + ['url={}'.format(store_url)]
+    ds.repo.init_remote('store', options=init_opts)
+
+    # copy files into the RIA store
+    ds.push('.', to='store')
+
+    # we should see the exact same annex object tree
+    dsgit_dir, archive_dir, dsobj_dir = \
+        get_layout_locations(1, store, ds.id)
+    store_objects = get_all_files(dsobj_dir)
+    local_objects = get_all_files(ds.pathobj / '.git' / 'annex' / 'objects')
+    assert_equal(len(store_objects), 4)
+
+    if not ds.repo.is_managed_branch():
+        # with managed branches the local repo uses hashdirlower instead
+        # TODO: However, with dataset layout version 1 this should therefore
+        #       work on adjusted branch the same way
+        # TODO: Wonder whether export-archive-ora should account for that and
+        #       rehash according to target layout.
+        assert_equal(sorted([p for p in store_objects]),
+                     sorted([p for p in local_objects])
+                     )
+
+        if not io.get_7z():
+            raise SkipTest("No 7z available in RIA store")
+
+        # we can simply pack up the content of the remote into a
+        # 7z archive and place it in the right location to get a functional
+        # archive remote
+
+        create_store(io, ppp_archiv_store, '1')
+        create_ds_in_store(io, ppp_archiv_store, ds.id, '2', '1')
+
+        whereis = ds.repo.whereis('one.txt')
+        dsgit_dir, archive_dir, dsobj_dir = \
+            get_layout_locations(1, archiv_store, ds.id)
+        ds.export_archive_ora(archive_dir / 'archive.7z')
+        init_opts = common_init_opts + ['url={}'.format(arch_url)]
+        ds.repo.init_remote('archive', options=init_opts)
+        # now fsck the new remote to get the new special remote indexed
+        ds.repo.fsck(remote='archive', fast=True)
+        assert_equal(len(ds.repo.whereis('one.txt')), len(whereis) + 1)
+        # test creating an archive with filters on files
+        ds.export_archive_ora(archive_dir / 'archive2.7z', annex_wanted='(include=*.txt)')
+        # test with wanted expression of a specific remote
+        ds.repo.set_preferred_content("wanted", "include=subdir/*", remote="store")
+        ds.export_archive_ora(archive_dir / 'archive3.7z', remote="store")
+        # test with the current sha
+        ds.export_archive_ora(
+            archive_dir / 'archive4.7z',
+            froms=ds.repo.get_revisions()[1],
```

Comment:
This parameter name has a typo, yet the tests pass. This appears to be an indication that this particular test implementation never runs.

Reply:
I think the archive is purposefully named `archive4.7z`. The test uses `archive2.7z`, `archive3.7z`, and `archive4.7z`. The original test seems incomplete. It just seems to check whether `export_archive_ora` succeeds and never checks the content of the created archives.

The test was executed and triggered errors before it was patched.

Reply:
Sorry, I was not precise enough. I was talking about `froms=`. AFAICS there is no such argument.

Reply:
I was surprised as well, but there is a `froms`-argument in `ExportArchiveORA`. It holds the content of the `--from` command line argument. Obviously the name was chosen to prevent using the keyword `from`. It should probably have been called `from_` though.

Reply:
Ah! :man_facepalming: Thanks for clarifying that!